evm/deploy-contract.js

```diff
@@ -100,43 +161,101 @@ async function deploy(options, chain) {
 
     const contractConfig = contracts[contractName];
     const constructorArgs = await getConstructorArgs(contractName, contracts);
-    const gasOptions = contractConfig.gasOptions || chain.gasOptions || null;
+    const gasOptions = contractConfig.gasOptions || chain.gasOptions || {};
     printInfo(`Constructor args for chain ${chain.name}`, constructorArgs);
     console.log(`Gas override for chain ${chain.name}: ${JSON.stringify(gasOptions)}`);
 
     const salt = options.salt || contractName;
-    printInfo('Contract deployment salt', salt);
+    let constAddressDeployer;
+    let create3Deployer;
+
+    switch (deployMethod) {
+        case 'create': {
+            const nonce = await provider.getTransactionCount(wallet.address);
```

Comment:
we're deploying two contracts, so proxy will use the next nonce
```suggestion
            const nonce = await provider.getTransactionCount(wallet.address) + 1;
```